src/OpenTelemetry.Shared/Guard.cs

```diff
@@ -0,0 +1,114 @@
+// <copyright file="Guard.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace OpenTelemetry.Shared
+{
+    public static class Guard
+    {
+        [DebuggerHidden]
```

Comment:
See how this tag affects debugging and decide if it should not be included for all methods in this class.

Reply:
From my experience, such helpers cause more issues than benefits, especially in open-source projects when not everyone may be aware of such "helpers".

Here is also a nice article on why it may not be the best to extract methods for such things: https://blog.ploeh.dk/2014/08/07/why-dry/

Probably we would also want to add `[MethodImpl(MethodImplOptions.AggressiveInlining)]` if we would like to merge this. However, let's keep in mind that this is an instrumentation library, and we should minimize the overhead. (especially when it is easy)

PS. Sorry for my English